lib/src/lifecycle_module.dart

```diff
@@ -321,11 +321,19 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
     }
 
     _state = LifecycleState.loading;
-    _transition = new Completer<Null>();
 
-    _load().then(_transition.complete).catchError(_transition.completeError);
+    // Keep track of this load's completer
+    final transition = new Completer<Null>();
 
-    return _transition.future;
+    // because this one can get overwritten
+    _transition = transition;
```

Comment:
Do we need to do this for the other transitions as well? I'm not seeing anything that would make this only an issue for `load`

Reply:
Added.